src/fu-util.c

```diff
@@ -2229,6 +2270,39 @@ fu_util_check_polkit_actions (GError **error)
 	return TRUE;
 }
 
+static gboolean
+fu_util_parse_filter_flags (FuUtilPrivate *priv, const gchar *filter, GError **error)
+{
+	g_auto(GStrv) strv = g_strsplit (filter, ",", -1);
+	for (guint i = 0; strv[i] != NULL; i++) {
+		FwupdDeviceFlags tmp;
+		if (g_str_has_prefix (strv[i], "~")) {
+			tmp = fwupd_device_flag_from_string (strv[i] + 1);
+			if (tmp == FWUPD_DEVICE_FLAG_UNKNOWN) {
+				g_set_error (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "Unknown device flag %s",
+					     strv[i] + 1);
+				return FALSE;
+			}
+			priv->filter_exclude |= tmp;
+		} else {
+			tmp = fwupd_device_flag_from_string (strv[i]);
+			if (tmp == FWUPD_DEVICE_FLAG_UNKNOWN) {
+				g_set_error (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "Unknown device flag %s",
+					     strv[i]);
+				return FALSE;
+			}
+			priv->filter_include |= tmp;
+		}
+	}
+	return TRUE;
```

Comment:
To avoid undefined behavior, I would think that it's worth putting an extra check here that `filter_exclude` and `filter_include` don't have any of the same flags included.

Reply:
Good idea, will fix.

Reply:
Done.